cmd/cost/get.go

```diff
@@ -0,0 +1,230 @@
+
+	//Populate accountSlice with accounts by looping until accounts.NextToken is null
+	for {
+		accounts, err := awsClient.ListAccountsForParent(&organizations.ListAccountsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: nextToken,
+		})
+		if err != nil {
+			log.Fatalln("Unable to retrieve accounts under OU:", err)
+		}
+
+		for i := 0; i < len(accounts.Accounts); i++ {
+			accountSlice = append(accountSlice, accounts.Accounts[i].Id)
+		}
+
+		if accounts.NextToken == nil {
+			break
+		}
+		nextToken = accounts.NextToken //If NextToken != nil, keep looping
+	}
+
+	return accountSlice
+}
+
+//Get the account IDs of all (not only immediate) accounts under OU
+func getAccountsRecursive(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) []*string {
+	var accountsIDs []*string
+
+	//Populate OUs
+	OUs := getOUs(OU, awsClient)
+
+	//Loop through all child OUs, get their costs, and store it to cost of current OU
+	for _, childOU := range OUs {
+		accountsIDs = append(accountsIDs, getAccountsRecursive(childOU, awsClient)...)
+	}
+
+	return append(accountsIDs, getAccounts(OU, awsClient)...)
+}
+
+//Get immediate OUs (child nodes) directly under given OU
+func getOUs(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) []*organizations.OrganizationalUnit {
+	var OUSlice []*organizations.OrganizationalUnit
+	var nextToken *string
+
+	//Populate OUSlice with OUs by looping until OUs.NextToken is null
+	for {
+		OUs, err := awsClient.ListOrganizationalUnitsForParent(&organizations.ListOrganizationalUnitsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: nextToken,
+		})
+		if err != nil {
+			log.Fatalln("Unable to retrieve child OUs under OU:", err)
+		}
+
+		//Add OUs to slice
+		for childOU := 0; childOU < len(OUs.OrganizationalUnits); childOU++ {
+			OUSlice = append(OUSlice, OUs.OrganizationalUnits[childOU])
+		}
+
+		if OUs.NextToken == nil {
+			break
+		}
+		nextToken = OUs.NextToken //If NextToken != nil, keep looping
+	}
+
+	return OUSlice
+}
+
+//Get the account IDs of all (not only immediate) accounts under OU
+func getOUsRecursive(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) []*organizations.OrganizationalUnit {
+	var OUs []*organizations.OrganizationalUnit
+
+	//Populate OUs by getting immediate OUs (direct nodes)
+	currentOUs := getOUs(OU, awsClient)
+
+	//Loop through all child OUs. Append the child OU, then append the OUs of the child OU
+	for _, currentOU := range currentOUs {
+		OUs = append(OUs, currentOU)
+		OUs = append(OUs, getOUsRecursive(currentOU, awsClient)...)
+	}
+
+	return OUs
+}
+
+//Get cost of given account
+func getAccountCost(accountID *string, awsClient awsprovider.Client, timePtr *string, cost *float64) {
+	//Starting from the 1st of the current month last year i.e. if today is 2020-06-29, then start date is 2019-06-01
+	start := strconv.Itoa(time.Now().Year()-1) + time.Now().Format("-01-") + "01"
+	end := time.Now().Format("2006-01-02")
```

Comment:
Why is this end date in 2006?

Reply:
For Go, the constants "2006" = Current Year, "01" = Current Month, and "02" = Current Day.

Reply:
Ah okay, I see its the format. Does AWS require this format?

Reply:
I believe so. For example, the AWS documentation gives the format for dates as “2017-05-01” and “2017-01-01”.

Reply:
Would you mind finding the documentation for the AWS requirement?

Reply:
Sorry, here's the link: https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_DateInterval.html